client/src/components/HabitForm/HabitForm.jsx

```diff
@@ -28,14 +33,22 @@ export class _HabitForm extends Component {
 	};
 
 	static propTypes = {
-		onSubmit: PropTypes.func.isRequired,
+		addHabit: PropTypes.func.isRequired,
 	};
 
 	handleChange = ({ target }) => {
 		const { name, value } = target;
 		this.setState({ [name]: value });
 	};
 
+	handleSubmit = () => {
+		const { steps, ...habit } = this.state;
+		habit.tags = habit.tags.map(t => t.text);
+
+		this.props.addHabit(habit);
+		this.props.history.push('/dashboard');
```

Comment:
I think this redirect should go in the habit action, if there are no errors. That way, if the form input is invalid the user will not be redirected to dashboard.